frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -67,43 +71,223 @@ export class Gallery extends React.Component {
-        console.log('%c<Gallery> Custom event', 'color: #28a745', type, data)
-        const newFolderId = qs.parse(data.config.history.location.search).folder_ids
-        if (data.config.appConfig.workspaceId !== state.config.appConfig.workspaceId || newFolderId !== state.folderId) {
-          this.setState({
-            config: data.config,
-            folderId: newFolderId
-          })
+    if (deletedIndex !== -1) {
+      imagesPreviews = state.imagesPreviews.slice(0, deletedIndex).concat(
+        state.imagesPreviews.slice(deletedIndex + 1)
+      )
+
+      // We set the new current index
+      if (deletedIndex < state.displayedPictureIndex) {
+        // if the currently displayed picture is after the deleted image
+        // we have to fix its index. The current picture's new index is
+        // decremented.
+        displayedPictureIndex--
+      } else if (deletedIndex === state.displayedPictureIndex) {
+        // if the currently displayed picture is the one being deleted
+        // we show the next picture, which index is the now the one of the
+        // deleted picture.
+
+        if (state.displayedPictureIndex >= imagesPreviews.length) {
+          // if this picture does not exist, though, we take the previous one.
+          // if no there are no pictures left, we take 0, which is the default
+          // value of displayedPictureIndex.
+          displayedPictureIndex = Math.max(0, state.displayedPictureIndex - 1)
         }
-        break
       }
-      case CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE:
-        console.log('%c<Gallery> Custom event', 'color: #28a745', type, data)
-        this.setState(prev => ({
-          loggedUser: {
-            ...prev.loggedUser,
-            lang: data
-          }
-        }))
-        i18n.changeLanguage(data)
-        this.buildBreadcrumbs()
-        if (state.workspaceLabel) this.setHeadTitle(`${props.t('Gallery')} · ${state.workspaceLabel}`)
-        break
-      default:
-        break
+    }
+
+    this.setState({ imagesPreviews, displayedPictureIndex })
+  }
+
+  setNewPicturesPreviews = imagesPreviews => {
+    // WARNING: modifies its parameter in the following line
+    imagesPreviews.sort((img1, img2) => img1.label.localeCompare(img2.label))
+
+    const displayedPictureId = this.displayedPictureId()
+
+    const displayedPictureIndex = Math.max(
+      0,
+      displayedPictureId
+        ? imagesPreviews.findIndex(image => image.contentId === displayedPictureId)
+        : 0
+    )
+
+    this.setState({ imagesPreviews, displayedPictureIndex })
+  }
+
+  handleShowApp = data => {
+    console.log('%c<Gallery> Custom event', 'color: #28a745', data)
+    const newFolderId = qs.parse(data.config.history.location.search).folder_ids
+    if (data.config.appConfig.workspaceId !== this.state.config.appConfig.workspaceId || newFolderId !== this.state.folderId) {
+      this.setState({
+        config: data.config,
+        folderId: newFolderId
```

Comment:
Check that the folderId is initialized somewhere

Reply:
ok, it is set in the constructor of the app. Let's just remove the irrelevant showApp custom event.